File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Arms/AlibabaCloud/SDK/ROS/CDK/Arms/AlertContactGroupProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Arms
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `ALIYUN::ARMS::AlertContactGroup`.</summary>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-arms.AlertContactGroupProps")]
    public class AlertContactGroupProps : AlibabaCloud.SDK.ROS.CDK.Arms.IAlertContactGroupProps
    {
        /// <summary>Property contactGroupName: The name of the alert contact group that you want to create.</summary>
        [JsiiProperty(name: "contactGroupName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOverride: true)]
        public object ContactGroupName
        {
            get;
            set;
        }

        /// <summary>Property contactIds: The list of alert contact ID.</summary>
        [JsiiProperty(name: "contactIds", typeJson: "{\"union\":{\"types\":[{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"},{\"collection\":{\"elementtype\":{\"union\":{\"types\":[{\"primitive\":\"number\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}},\"kind\":\"array\"}}]}}", isOverride: true)]
        public object ContactIds
        {
            get;
            set;
        }

        /// <summary>Property proxyUserId: Internal parameters.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "proxyUserId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true, isOverride: true)]
        public object? ProxyUserId
        {
            get;
            set;
        }

        /// <summary>Property regionId: Region ID.</summary>
        /// <remarks>
        /// Default to region of stack.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "regionId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true, isOverride: true)]
        public object? RegionId
        {
            get;
            set;
        }
    }
}

File: multiple-languages/go/alicloudroscdkarms/AlertContactGroupProps.go
package alicloudroscdkarms


// Properties for defining a `AlertContactGroup`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-arms-alertcontactgroup
type AlertContactGroupProps struct {
	// Property contactGroupName: The name of the alert contact group that you want to create.
	ContactGroupName interface{} `field:"required" json:"contactGroupName" yaml:"contactGroupName"`
	// Property contactIds: The list of alert contact ID.
	ContactIds interface{} `field:"required" json:"contactIds" yaml:"contactIds"`
	// Property proxyUserId: Internal parameters.
	ProxyUserId interface{} `field:"optional" json:"proxyUserId" yaml:"proxyUserId"`
	// Property regionId: Region ID.
	//
	// Default to region of stack.
	RegionId interface{} `field:"optional" json:"regionId" yaml:"regionId"`
}

